### ISSUE TITLE
Explicit agent saving

### ISSUE THREAD

**Issue by slundell** [reinforceio/tensorforce#204]
There needs to be a way to explicitly save an agent. For instance when you check test data performance in episode_finnished, you might want to save the agent/model every time it performs better than before in the training process.

**Comment by AlexKuhnle** [comment]
The (renamed) methods `Model.save` and `Model.restore` work again with the new commit, hence also `Agent.save_model`, with updated API. Hope that addresses the problem.

**Comment by slerman12** [comment]
Hi, I'm also having an issue with this. I set the saving directory to "./saved/" and then tried loading from there, but received and error. Here's the error with one of the provided examples:

`import numpy as np
from tensorforce.agents import PPOAgent
from tensorforce.execution import Runner
from tensorforce.contrib.openai_gym import OpenAIGym

# Create an OpenAIgym environment
env = OpenAIGym('CartPole-v0')

# Network as list of layers
network_spec = [
    dict(type='dense', size=32, activation='tanh'),
    dict(type='dense', size=32, activation='tanh')
]

agent = PPOAgent(
    states_spec=env.states,
    actions_spec=env.actions,
    network_spec=network_spec,
    batch_size=4096,
    # Agent
    preprocessing=None,
    exploration=None,
    reward_preprocessing=None,
    # BatchAgent
    keep_last_timestep=True,
    # PPOAgent
    step_optimizer=dict(
        type='adam',
        learning_rate=1e-3
    ),
    optimization_steps=10,
    # Model
    scope='ppo',
    discount=0.99,
    # DistributionModel
    distributions_spec=None,
    entropy_regularization=0.01,
    # PGModel
    baseline_mode=None,
    baseline=None,
    baseline_optimizer=None,
    gae_lambda=None,
    normalize_rewards=False,
    # PGLRModel
    likelihood_ratio_clipping=0.2,
    summary_spec=None,
    distributed_spec=None
)

# Create the runner
runner = Runner(agent=agent, environment=env)


# Callback function printing episode statistics
def episode_finished(r):
    print("Finished episode {ep} after {ts} timesteps (reward: {reward})".format(ep=r.episode, ts=r.episode_timestep,
                                                                                 reward=r.episode_rewards[-1]))

    if r.episode == 90:
        r.agent.save_model(directory="./saved/")
    return True


# Start learning
runner.run(episodes=100, max_episode_timesteps=200, episode_finished=episode_finished)

# Print statistics
print("Learning finished. Total episodes: {ep}. Average reward of last 100 episodes: {ar}.".format(
    ep=runner.episode,
    ar=np.mean(runner.episode_rewards[-100:]))
)

saved_agent = agent.restore_model(directory="./saved/")
env = gym.make("CartPole-v0")
s = env.reset()
done = False
while not done:
    env.render()
    action = saved_agent.act(s)
    s, r, done, _ = env.step(action)
`
